pipeline/backend/kubernetes/pod.go

```diff
@@ -505,6 +514,18 @@ func startPod(ctx context.Context, engine *kube, step *types.Step, options Backe
 		return nil, err
 	}
 
+	if needsImagePullSecret(step) {
```

Comment:
And probably call it in `kubernetes.go`. See how volumes work.

Reply:
Created startRegistrySecret and stopRegistrySecret to mirror startPod/Volume and stopPod/Volume